infra/core/security/documentdb-sql-role.bicep

```diff
@@ -0,0 +1,27 @@
+metadata name = 'DocumentDB Database Account SQL Role Assignments.'
+metadata description = 'Assign RBAC role for data plane access to Azure Cosmos DB for NoSQL.'
+
+@description('Name of the Azure Cosmos DB for NoSQL account.')
+param databaseAccountName string
+
+@description('Id of the identity/principal to assign this role in the context of the account.')
+param principalId string = ''
+
+@description('Id of the role definition to assign to the targeted principal in the context of the account.')
+param roleDefinitionId string
+
+resource databaseAccount 'Microsoft.DocumentDB/databaseAccounts@2023-04-15' existing = {
+  name: databaseAccountName
+}
+
+resource sqlRoleAssignment 'Microsoft.DocumentDB/databaseAccounts/sqlRoleAssignments@2023-04-15' = {
```

Comment:
I assume this wasn't available in the avm module?

Reply:
This SQL Role setting can be created together with the AVM that creates the CosmosDB resource, but because a circular reference occurred, I had no choice but to externalize the SQL Role. Specifically, the following situation arose:

- To create CosmosDB, we need the principalId of the AppService to grant a role that can read and write to the database, so the AppService needs to be deployed first.

- The AppService module needs to store the CosmosDB access destination in an environment variable, so CosmosDB needed to be deployed first.

This required the module that sets the SQL Role to be separated, but I couldn't find an AVM that only sets the SQL Role, so I defined it separately.

Reply:
Okay, yep, that makes sense, thanks for clarifying!